### PR TITLE
feat: extending content negotiation

### PR DESCRIPTION
**Pull request by rflechtner**
## NO-TICKET
Our Kilt driver so far does not handle the content types consistently; it returns an incomplete version of MIME type `application/ld+json;profile="https://w3id.org/did-resolution"`, but marked it as `application/did+json` or `application/did+ld+json`. 
This PR changes this and makes use of express's own content negotiation functionality.

1. If the accept header is set to `application/(did+)json` or `application/did+ld+json`, the driver will now only return a did document, either in its json or json-ld representation. The return type is indicated as `application/did+(ld+)json`.
2. If the accept header is blank or set to `application/ld+json;profile="https://w3id.org/did-resolution"` (or equivalently `application/did+ld+json`) you will get the full did resolution result containing resolution and document metadata. This includes error cases, where the error is signalled in the resolution metadata and via the http response code. The return MIME type is `application/ld+json;profile="https://w3id.org/did-resolution"`.
3. In case the resolution produces an error, depending on the requested MIME type you either get a resolution result where the did document is set to null, or you get an empty response, with the status code signalling the error.
4. Querying a deactivated did gives you a 410 response code now, but still returns the did document stub as before. 

See https://w3c-ccg.github.io/did-resolution/#did-resolution-result & https://w3c-ccg.github.io/did-resolution/#bindings-https for the specs concerning did resolver results.

## How to test:
Use with universal resolver or make curl requests to it, using different content headers.

## Checklist:

- [x] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

**Review by Diiaablo95 — COMMENTED** [review]
I added some minor comments, but I also want to discuss about a potential bigger issue. When running behind the uniresolver proxy, the result is now always a JSON-LD, even thought the return type is `application/did+json`. I think this is not expected, and I presume that whenever the proxy finds some json-ld info, like a `@context` or something, it transforms the result to a JSON-LD result.

File: src/index.js [hunk 1]
```diff
@@ -21,65 +25,108 @@ async function start() {
...
+        console.info(JSON.stringify(req.headers, null, 2))
+        const { did } = req.params
 
-      if (!didResolutionResult) {
-        console.trace(`\n🔍 DID ${did} not found (on chain)`)
-        res.sendStatus(404)
-        return
-      }
+        const didResolutionResult = {
+          '@context': DID_RESOLUTION_RESPONSE_CONTEXT,
```
[the remark this answers is not in the document]
Comment (rflechtner) [reply src/index.js]: https://github.com/KILTprotocol/kilt-did-driver/pull/24/commits/edb1da7717305256547b1035bfbcf30948c33898

File: src/index.js [hunk 2]
```diff
@@ -21,65 +25,108 @@ async function start() {
...
+        // which is represented by the sole `id` property.
+        // https://www.w3.org/TR/did-core/#did-document-properties
+        if (didResolutionResult.didDocumentMetadata.deactivated) {
+          // sending a 410 according to https://w3c-ccg.github.io/did-resolution/#bindings-https
+          res.status(410)
+          didResolutionResult.didDocument = {
+            id: did
+          }
+          if (isJsonLd) {
+            didResolutionResult.didDocument['@context'] = DID_DOC_CONTEXT
```
[the remark this answers is not in the document]
Comment (rflechtner) [reply src/index.js]: https://github.com/KILTprotocol/kilt-did-driver/pull/24/commits/edb1da7717305256547b1035bfbcf30948c33898

File: src/consts.js [hunk 3]
```diff
@@ -8,6 +8,15 @@
 // did
 const URI_DID = '/1.0/identifiers/:did'
 
+const DID_RESOLUTION_RESPONSE_MIME =
+  'application/ld+json;profile="https://w3id.org/did-resolution"'
+
+const DID_DOC_CONTEXT = ['https://www.w3.org/ns/did/v1']
+const DID_RESOLUTION_RESPONSE_CONTEXT = ['https://w3id.org/did-resolution/v1']
```
[the remark this answers is not in the document]
Comment (rflechtner) [reply src/consts.js]: https://github.com/KILTprotocol/kilt-did-driver/pull/24/commits/edb1da7717305256547b1035bfbcf30948c33898

File: src/consts.js [hunk 4]
```diff
@@ -8,6 +8,15 @@
 // did
 const URI_DID = '/1.0/identifiers/:did'
 
+const DID_RESOLUTION_RESPONSE_MIME =
+  'application/ld+json;profile="https://w3id.org/did-resolution"'
+
+const DID_DOC_CONTEXT = ['https://www.w3.org/ns/did/v1']
```
[the remark this answers is not in the document]
Comment (rflechtner) [reply src/consts.js]: https://github.com/KILTprotocol/kilt-did-driver/pull/24/commits/edb1da7717305256547b1035bfbcf30948c33898

**Pull request merged**